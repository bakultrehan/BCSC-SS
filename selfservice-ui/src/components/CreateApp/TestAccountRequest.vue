/** * TestAccountRequest component */

<template>
  <v-card class="mx-auto outer-card">
    <v-toolbar flat class="bc-subtitle padding-0" dark>
      <v-btn icon @click="goBack()" :aria-label="$t('testAccount.btnBack')">
        <v-icon>mdi-arrow-left</v-icon>
      </v-btn>
      <h1 class="bc-h1-sub-ttile">{{ $t('testAccount.pagetitle') }}</h1>
      <div class="flex-grow-1"></div>
    </v-toolbar>
    <v-divider></v-divider>
    <v-item-group mandatory :value="slectedNumber">
      <v-container>
        <v-row class="mx-4">
          <v-col cols="12" class="pb-0" flat>
            <Alert type="error" v-if="errorStatus" class="alert-top">Something went wrong...</Alert>
            <v-card flat class="pb-0">
              <!-- <v-list-item-content>BCSC Test Account</v-list-item-content> -->
              <v-list-item-content v-html="$t('testAccount.pageinfo', { package: 'package' })"></v-list-item-content>
            </v-card>
          </v-col>

          <v-col cols="12" class="pt-0" flat>
            <v-card flat>
              <v-list-item-content>{{ $t('testAccount.how_many_test_account') }}</v-list-item-content>
            </v-card>
          </v-col>

          <v-col v-for="(testAccount, idx) in noOfTestAccounts" :key="idx" class="card-width">
            <v-item v-slot:default="{ active }" :value="testAccount" class="test-account">
              <v-card
                class="d-flex align-center pa-4 test-account"
                :class="active ? 'active-bg' : ''"
                @click="selectedTestAccount(testAccount)"
                @keyup.enter="selectedTestAccount(testAccount)"
                :data-test-id="`input-select-test-account-${testAccount}`"
              >
                <v-list-item>
                  <v-list-item-content class="text-center">
                    <v-list-item-title class="headline">{{ testAccount }}</v-list-item-title>
                    <v-list-item-subtitle></v-list-item-subtitle>
                  </v-list-item-content>
                </v-list-item>
                <v-spacer></v-spacer>
              </v-card>
            </v-item>
          </v-col>
          <v-col cols="12" flat>
            <v-card flat>
              <v-list-item-content>{{ $t('testAccount.special_notes') }}</v-list-item-content>
              <v-list-item-content class="subtitle-1" v-html="$t('testAccount.specialNotesInfo')"></v-list-item-content>
              <!-- <TextArea
                v-model="notes"
                :label="$t('testAccount.special_notes')"
                type="text"
                outlined
              />-->
            </v-card>
          </v-col>
        </v-row>
      </v-container>
    </v-item-group>
    <v-col cols="12">
      <v-card flat>
        <v-divider></v-divider>
        <v-card-actions class="mx-4">
          <v-spacer></v-spacer>
          <Button
            @click="goBack()"
            @keyup.enter="goBack()"
            :aria-label="$t('testAccount.btnBack')"
            secondary
            data-test-id="btn-cancel-test-account"
          >
            {{
            $t(
            showWizardExperience()
            ? 'testAccount.btnBack'
            : 'testAccount.btnCancel'
            )
            }}
          </Button>
          <Button
            :disabled="!slectedNumber"
            :loading="isLoading"
            class="white--text submit-account ml-6"
            depressed
            @click="submitTestAccount"
            @keyup.enter="submitTestAccount"
            data-test-id="btn-submit-test-account"
          >
            {{
            $t(
            showWizardExperience()
            ? 'testAccount.btnNext'
            : 'testAccount.btnSaveChanges'
            )
            }}
          </Button>
        </v-card-actions>
      </v-card>
    </v-col>
  </v-card>
</template>
<script lang="ts">
import { Component, Vue, Watch, Prop } from 'vue-property-decorator';
import { Getter, namespace, Action } from 'vuex-class';
import Button from '@/Atomic/Button/Button.vue';
import TextArea from '@/Atomic/TextArea/TextArea.vue';
import Alert from '@/Atomic/Alert/Alert.vue';

const PackageAndTestModule = namespace('PackageAndTestModule');
const TechnicalReqModule = namespace('TechnicalReqModule');
const SharedModule = namespace('SharedModule');

@Component({
  components: {
    Button,
    TextArea,
    Alert
  }
})
export default class TestAccountRequest extends Vue {
  @Prop({ default: 0 })
  public id!: number;
  @PackageAndTestModule.Getter('successStatus') public successStatus!: boolean;
  @PackageAndTestModule.Getter('errorStatus') public errorStatus!: boolean;
  @PackageAndTestModule.Action('loadPackage') public loadPackage!: any;
  @PackageAndTestModule.Getter('getPackageList') public getPackageList!: [];
  @PackageAndTestModule.Action('clearStatus') public clearStatus!: any;
  @PackageAndTestModule.Action('addTestAccountRequestToProject')
  public addTestAccountRequestToProject!: any;

  @TechnicalReqModule.Action('loadTechnicalReqDetails')
  public loadTechnicalReqDetails!: any;
  @TechnicalReqModule.Getter('getTechnicalReq')
  public getTechnicalReq!: any;
  @TechnicalReqModule.Getter('isLoading') public isLoading!: boolean;

  @SharedModule.Action('redirectFromSummaryPage')
  public redirectFromSummaryPage!: any;
  @SharedModule.Getter('isRedirectFromSummaryPage')
  public isRedirectFromSummaryPage!: boolean;

  private noOfTestAccounts: any = [1, 2, 3, 5];
  private notes: string = '';

  private slectedNumber: number = 1;
  // private isLoading: boolean = false;
  private projectId: number = this.id || 0;

  @Watch('successStatus')
  private onStatusChanged(val: any, oldVal: any) {
    setTimeout(this.clearStatus, 3000);
  }
  @Watch('getTechnicalReq')
  private ongetTechnicalReqChanged(val: any) {
    this.slectedNumber = val.noOfTestAccount || this.slectedNumber;
    this.notes = val.noteTestAccount || this.notes;
  }

  private selectedTestAccount(packageVal: number) {
    this.slectedNumber = packageVal;
  }
  private submitTestAccount() {
    // add package to project com ehere
    this.addTestAccountRequestToProject({
      noOfTestAccount: this.slectedNumber,
      noteTestAccount: this.notes,
      projectId: this.projectId
    });
  }

  private mounted() {
    if (this.id !== 0) {
      // this.isEditMode = true;
      this.loadTechnicalReqDetails(this.id);
    }
  }

  private goBack() {
    const redirectPage = this.showWizardExperience()
      ? `/project/${this.projectId}/package/`
      : `/project-container/${this.projectId}/`;

    this.redirectFromSummaryPage(false);
    this.$router.push(redirectPage);
  }
  private showWizardExperience() {
    if (this.isRedirectFromSummaryPage) {
      return false;
    }
    return true;
  }
}
</script>

<style lang="scss" scoped>
@import './../../assets/styles/theme.scss';
.active-bg {
  background-color: $BCgovActiveBg;
}
.card-width {
  max-width: 110px;
}
.text-center {
  text-align: center !important;
}
</style>
