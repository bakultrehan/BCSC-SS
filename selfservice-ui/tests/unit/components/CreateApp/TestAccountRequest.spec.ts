import { mount, createLocalVue } from '@vue/test-utils';
import TestAccountRequest from '@/components/CreateApp/TestAccountRequest.vue';
import Vuetify from 'vuetify';
import Vuex from 'vuex';
import i18n from '@/i18n';
// import Vue from 'vue';

describe('TestAccountRequest.vue', () => {
  let vuetify: any;
  const localVue = createLocalVue();
  // tslint:disable-next-line
  const $t = jest.fn(() => {});

  localVue.use(Vuetify);
  localVue.use(Vuex);
  vuetify = new Vuetify();
  const store = new Vuex.Store({
    modules: {
      PackageAndTestModule: {
        namespaced: true,
        state: {},
        getters: {
          isLoggedin: jest.fn(),
          successStatus: jest.fn(),
          errorStatus: jest.fn()
        },
        actions: {
          addTestAccountRequestToProject: jest.fn()
        }
      }
    }
  });

  const mountFunction = (options: any) => {
    return mount(TestAccountRequest, {
      store,
      vuetify,
      localVue,
      mocks: { $t },
      ...options
    });
  };

  it('renders props when passed', () => {
    const TestReqWrapper = mountFunction({});

    expect(TestReqWrapper.element).toMatchSnapshot();
  });

  it('select on package on click ', () => {
    const TestReqWrapper = mountFunction({});
    const event = jest.fn();
    const button = TestReqWrapper.find('.test-account');
    TestReqWrapper.vm.$on('action-btn:clicked', event);
    button.trigger('click');

    expect(TestReqWrapper.element).toMatchSnapshot();
  });

  it('submit form with selected package on submit click ', () => {
    const TestReqWrapper = mountFunction({});
    const event = jest.fn();
    const button = TestReqWrapper.find('.submit-account');
    TestReqWrapper.vm.$on('action-btn:clicked', event);
    button.trigger('click');

    expect(TestReqWrapper.element).toMatchSnapshot();
  });
});
