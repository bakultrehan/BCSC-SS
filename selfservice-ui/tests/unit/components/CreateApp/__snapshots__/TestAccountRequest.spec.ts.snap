// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestAccountRequest.vue renders props when passed 1`] = `
<div
  class="mx-auto outer-card v-card v-sheet theme--light"
>
  <header
    class="bc-subtitle padding-0 v-sheet v-sheet--tile theme--dark v-toolbar v-toolbar--flat"
    style="height: 64px;"
  >
    <div
      class="v-toolbar__content"
      style="height: 64px;"
    >
      <button
        class="v-btn v-btn--flat v-btn--icon v-btn--round theme--dark v-size--default"
        type="button"
      >
        <span
          class="v-btn__content"
        >
          <i
            aria-hidden="true"
            class="v-icon notranslate mdi mdi-arrow-left theme--dark"
          />
        </span>
      </button>
       
      <div
        class="v-toolbar__title"
      >
        
      </div>
       
      <div
        class="flex-grow-1"
      />
       
      <div
        class="col-lg-4 col-md-5 col-8 col"
      >
        <!---->
      </div>
       
      <div
        class="flex-grow-1"
      />
    </div>
  </header>
   
  <hr
    aria-orientation="horizontal"
    class="v-divider theme--light"
    role="separator"
  />
   
  <div
    class="v-item-group theme--light"
  >
    <div
      class="container"
    >
      <div
        class="row mx-4"
      >
        <div
          class="col col-12"
          flat=""
        >
          <div
            class="v-card v-card--flat v-sheet theme--light"
          >
            <div
              class="v-list-item__content"
            />
          </div>
        </div>
         
        <div
          class="col col-12"
          flat=""
        >
          <div
            class="v-card v-card--flat v-sheet theme--light"
          >
            <div
              class="v-list-item__content"
            >
              
              
            
            </div>
          </div>
        </div>
         
        <div
          class="card-width col"
        >
          <div
            class="d-flex align-center pa-4 test-account test-account v-card v-card--link v-sheet theme--light active-bg"
            tabindex="0"
          >
            <div
              class="v-list-item theme--light"
              tabindex="-1"
            >
              <div
                class="v-list-item__content text-center"
              >
                <div
                  class="v-list-item__title headline"
                >
                  
                    1
                  
                </div>
                 
                <div
                  class="v-list-item__subtitle"
                />
              </div>
            </div>
             
            <div
              class="spacer"
            />
          </div>
        </div>
        <div
          class="card-width col"
        >
          <div
            class="d-flex align-center pa-4 test-account test-account v-card v-card--link v-sheet theme--light"
            tabindex="0"
          >
            <div
              class="v-list-item theme--light"
              tabindex="-1"
            >
              <div
                class="v-list-item__content text-center"
              >
                <div
                  class="v-list-item__title headline"
                >
                  
                    2
                  
                </div>
                 
                <div
                  class="v-list-item__subtitle"
                />
              </div>
            </div>
             
            <div
              class="spacer"
            />
          </div>
        </div>
        <div
          class="card-width col"
        >
          <div
            class="d-flex align-center pa-4 test-account test-account v-card v-card--link v-sheet theme--light"
            tabindex="0"
          >
            <div
              class="v-list-item theme--light"
              tabindex="-1"
            >
              <div
                class="v-list-item__content text-center"
              >
                <div
                  class="v-list-item__title headline"
                >
                  
                    3
                  
                </div>
                 
                <div
                  class="v-list-item__subtitle"
                />
              </div>
            </div>
             
            <div
              class="spacer"
            />
          </div>
        </div>
        <div
          class="card-width col"
        >
          <div
            class="d-flex align-center pa-4 test-account test-account v-card v-card--link v-sheet theme--light"
            tabindex="0"
          >
            <div
              class="v-list-item theme--light"
              tabindex="-1"
            >
              <div
                class="v-list-item__content text-center"
              >
                <div
                  class="v-list-item__title headline"
                >
                  
                    5
                  
                </div>
                 
                <div
                  class="v-list-item__subtitle"
                />
              </div>
            </div>
             
            <div
              class="spacer"
            />
          </div>
        </div>
         
        <div
          class="col col-12"
          flat=""
        >
          <div
            class="v-card v-card--flat v-sheet theme--light"
          >
            <div
              class="v-list-item__content"
            >
              
              
            
            </div>
             
            <div
              class="v-list-item__content subtitle-1"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="col col-12"
  >
    <div
      class="v-card v-card--flat v-sheet theme--light"
    >
      <hr
        aria-orientation="horizontal"
        class="v-divider theme--light"
        role="separator"
      />
       
      <div
        class="v-card__actions mx-4"
      >
        <div
          class="spacer"
        />
         
        <button
          class="btn v-btn v-btn--contained theme--light v-size--default secondary-btn normal"
          name="button"
          type="button"
        >
          <span
            class="v-btn__content"
          >
            
          
        
          </span>
        </button>
         
        <button
          class="btn white--text submit-account ml-6 v-btn v-btn--depressed theme--light v-size--default primary normal"
          depressed=""
          name="button"
          type="button"
        >
          <span
            class="v-btn__content"
          >
            
          
        
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`TestAccountRequest.vue select on package on click  1`] = `
<div
  class="mx-auto outer-card v-card v-sheet theme--light"
>
  <header
    class="bc-subtitle padding-0 v-sheet v-sheet--tile theme--dark v-toolbar v-toolbar--flat"
    style="height: 64px;"
  >
    <div
      class="v-toolbar__content"
      style="height: 64px;"
    >
      <button
        class="v-btn v-btn--flat v-btn--icon v-btn--round theme--dark v-size--default"
        type="button"
      >
        <span
          class="v-btn__content"
        >
          <i
            aria-hidden="true"
            class="v-icon notranslate mdi mdi-arrow-left theme--dark"
          />
        </span>
      </button>
       
      <div
        class="v-toolbar__title"
      >
        
      </div>
       
      <div
        class="flex-grow-1"
      />
       
      <div
        class="col-lg-4 col-md-5 col-8 col"
      >
        <!---->
      </div>
       
      <div
        class="flex-grow-1"
      />
    </div>
  </header>
   
  <hr
    aria-orientation="horizontal"
    class="v-divider theme--light"
    role="separator"
  />
   
  <div
    class="v-item-group theme--light"
  >
    <div
      class="container"
    >
      <div
        class="row mx-4"
      >
        <div
          class="col col-12"
          flat=""
        >
          <div
            class="v-card v-card--flat v-sheet theme--light"
          >
            <div
              class="v-list-item__content"
            />
          </div>
        </div>
         
        <div
          class="col col-12"
          flat=""
        >
          <div
            class="v-card v-card--flat v-sheet theme--light"
          >
            <div
              class="v-list-item__content"
            >
              
              
            
            </div>
          </div>
        </div>
         
        <div
          class="card-width col"
        >
          <div
            class="d-flex align-center pa-4 test-account test-account v-card v-card--link v-sheet theme--light active-bg"
            tabindex="0"
          >
            <div
              class="v-list-item theme--light"
              tabindex="-1"
            >
              <div
                class="v-list-item__content text-center"
              >
                <div
                  class="v-list-item__title headline"
                >
                  
                    1
                  
                </div>
                 
                <div
                  class="v-list-item__subtitle"
                />
              </div>
            </div>
             
            <div
              class="spacer"
            />
          </div>
        </div>
        <div
          class="card-width col"
        >
          <div
            class="d-flex align-center pa-4 test-account test-account v-card v-card--link v-sheet theme--light"
            tabindex="0"
          >
            <div
              class="v-list-item theme--light"
              tabindex="-1"
            >
              <div
                class="v-list-item__content text-center"
              >
                <div
                  class="v-list-item__title headline"
                >
                  
                    2
                  
                </div>
                 
                <div
                  class="v-list-item__subtitle"
                />
              </div>
            </div>
             
            <div
              class="spacer"
            />
          </div>
        </div>
        <div
          class="card-width col"
        >
          <div
            class="d-flex align-center pa-4 test-account test-account v-card v-card--link v-sheet theme--light"
            tabindex="0"
          >
            <div
              class="v-list-item theme--light"
              tabindex="-1"
            >
              <div
                class="v-list-item__content text-center"
              >
                <div
                  class="v-list-item__title headline"
                >
                  
                    3
                  
                </div>
                 
                <div
                  class="v-list-item__subtitle"
                />
              </div>
            </div>
             
            <div
              class="spacer"
            />
          </div>
        </div>
        <div
          class="card-width col"
        >
          <div
            class="d-flex align-center pa-4 test-account test-account v-card v-card--link v-sheet theme--light"
            tabindex="0"
          >
            <div
              class="v-list-item theme--light"
              tabindex="-1"
            >
              <div
                class="v-list-item__content text-center"
              >
                <div
                  class="v-list-item__title headline"
                >
                  
                    5
                  
                </div>
                 
                <div
                  class="v-list-item__subtitle"
                />
              </div>
            </div>
             
            <div
              class="spacer"
            />
          </div>
        </div>
         
        <div
          class="col col-12"
          flat=""
        >
          <div
            class="v-card v-card--flat v-sheet theme--light"
          >
            <div
              class="v-list-item__content"
            >
              
              
            
            </div>
             
            <div
              class="v-list-item__content subtitle-1"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="col col-12"
  >
    <div
      class="v-card v-card--flat v-sheet theme--light"
    >
      <hr
        aria-orientation="horizontal"
        class="v-divider theme--light"
        role="separator"
      />
       
      <div
        class="v-card__actions mx-4"
      >
        <div
          class="spacer"
        />
         
        <button
          class="btn v-btn v-btn--contained theme--light v-size--default secondary-btn normal"
          name="button"
          type="button"
        >
          <span
            class="v-btn__content"
          >
            
          
        
          </span>
        </button>
         
        <button
          class="btn white--text submit-account ml-6 v-btn v-btn--depressed theme--light v-size--default primary normal"
          depressed=""
          name="button"
          type="button"
        >
          <span
            class="v-btn__content"
          >
            
          
        
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`TestAccountRequest.vue submit form with selected package on submit click  1`] = `
<div
  class="mx-auto outer-card v-card v-sheet theme--light"
>
  <header
    class="bc-subtitle padding-0 v-sheet v-sheet--tile theme--dark v-toolbar v-toolbar--flat"
    style="height: 64px;"
  >
    <div
      class="v-toolbar__content"
      style="height: 64px;"
    >
      <button
        class="v-btn v-btn--flat v-btn--icon v-btn--round theme--dark v-size--default"
        type="button"
      >
        <span
          class="v-btn__content"
        >
          <i
            aria-hidden="true"
            class="v-icon notranslate mdi mdi-arrow-left theme--dark"
          />
        </span>
      </button>
       
      <div
        class="v-toolbar__title"
      >
        
      </div>
       
      <div
        class="flex-grow-1"
      />
       
      <div
        class="col-lg-4 col-md-5 col-8 col"
      >
        <!---->
      </div>
       
      <div
        class="flex-grow-1"
      />
    </div>
  </header>
   
  <hr
    aria-orientation="horizontal"
    class="v-divider theme--light"
    role="separator"
  />
   
  <div
    class="v-item-group theme--light"
  >
    <div
      class="container"
    >
      <div
        class="row mx-4"
      >
        <div
          class="col col-12"
          flat=""
        >
          <div
            class="v-card v-card--flat v-sheet theme--light"
          >
            <div
              class="v-list-item__content"
            />
          </div>
        </div>
         
        <div
          class="col col-12"
          flat=""
        >
          <div
            class="v-card v-card--flat v-sheet theme--light"
          >
            <div
              class="v-list-item__content"
            >
              
              
            
            </div>
          </div>
        </div>
         
        <div
          class="card-width col"
        >
          <div
            class="d-flex align-center pa-4 test-account test-account v-card v-card--link v-sheet theme--light active-bg"
            tabindex="0"
          >
            <div
              class="v-list-item theme--light"
              tabindex="-1"
            >
              <div
                class="v-list-item__content text-center"
              >
                <div
                  class="v-list-item__title headline"
                >
                  
                    1
                  
                </div>
                 
                <div
                  class="v-list-item__subtitle"
                />
              </div>
            </div>
             
            <div
              class="spacer"
            />
          </div>
        </div>
        <div
          class="card-width col"
        >
          <div
            class="d-flex align-center pa-4 test-account test-account v-card v-card--link v-sheet theme--light"
            tabindex="0"
          >
            <div
              class="v-list-item theme--light"
              tabindex="-1"
            >
              <div
                class="v-list-item__content text-center"
              >
                <div
                  class="v-list-item__title headline"
                >
                  
                    2
                  
                </div>
                 
                <div
                  class="v-list-item__subtitle"
                />
              </div>
            </div>
             
            <div
              class="spacer"
            />
          </div>
        </div>
        <div
          class="card-width col"
        >
          <div
            class="d-flex align-center pa-4 test-account test-account v-card v-card--link v-sheet theme--light"
            tabindex="0"
          >
            <div
              class="v-list-item theme--light"
              tabindex="-1"
            >
              <div
                class="v-list-item__content text-center"
              >
                <div
                  class="v-list-item__title headline"
                >
                  
                    3
                  
                </div>
                 
                <div
                  class="v-list-item__subtitle"
                />
              </div>
            </div>
             
            <div
              class="spacer"
            />
          </div>
        </div>
        <div
          class="card-width col"
        >
          <div
            class="d-flex align-center pa-4 test-account test-account v-card v-card--link v-sheet theme--light"
            tabindex="0"
          >
            <div
              class="v-list-item theme--light"
              tabindex="-1"
            >
              <div
                class="v-list-item__content text-center"
              >
                <div
                  class="v-list-item__title headline"
                >
                  
                    5
                  
                </div>
                 
                <div
                  class="v-list-item__subtitle"
                />
              </div>
            </div>
             
            <div
              class="spacer"
            />
          </div>
        </div>
         
        <div
          class="col col-12"
          flat=""
        >
          <div
            class="v-card v-card--flat v-sheet theme--light"
          >
            <div
              class="v-list-item__content"
            >
              
              
            
            </div>
             
            <div
              class="v-list-item__content subtitle-1"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="col col-12"
  >
    <div
      class="v-card v-card--flat v-sheet theme--light"
    >
      <hr
        aria-orientation="horizontal"
        class="v-divider theme--light"
        role="separator"
      />
       
      <div
        class="v-card__actions mx-4"
      >
        <div
          class="spacer"
        />
         
        <button
          class="btn v-btn v-btn--contained theme--light v-size--default secondary-btn normal"
          name="button"
          type="button"
        >
          <span
            class="v-btn__content"
          >
            
          
        
          </span>
        </button>
         
        <button
          class="btn white--text submit-account ml-6 v-btn v-btn--depressed theme--light v-size--default primary normal"
          depressed=""
          name="button"
          type="button"
        >
          <span
            class="v-btn__content"
          >
            
          
        
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
